concent_api/core/subtask_helpers.py

```diff
@@ -143,22 +144,25 @@ def update_all_clients_timed_out_subtasks(client_public_key: bytes) -> None:
 
     encoded_client_public_key = b64encode(client_public_key)
 
-    clients_subtask_list = Subtask.objects.select_for_update().filter(
+    clients_subtask_list = Subtask.objects.filter(
         Q(requestor__public_key=encoded_client_public_key) | Q(provider__public_key = encoded_client_public_key),
         state__in=[state.name for state in Subtask.ACTIVE_STATES],
     )
 
     for subtask in clients_subtask_list:
-        if subtask.state_enum == Subtask.SubtaskState.FORCING_RESULT_TRANSFER and subtask.requestor.public_key_bytes == client_public_key:
-            verify_file_status(subtask=subtask)
+        with transaction.atomic(using='control'):
+            Subtask.objects.select_for_update().filter(subtask_id=subtask.subtask_id)
```

Comment:
In issue is described, that every subtask should be processed in separate transaction:

>  It should first find all subtasks that may need processing. Then process timeouts for each one in a separate transaction. We need a commit after each subtask so that a failure for one does not invalidate changes already made for another one.